firstvoices/backend/serializers/async_results_serializers.py

```diff
@@ -0,0 +1,51 @@
+from collections import OrderedDict
+
+from celery.result import AsyncResult
+from rest_framework import serializers
+
+from backend.models.async_results import CustomOrderRecalculationPreviewResult
+
+
+class CustomOrderRecalculationPreviewResultDetailSerializer(
+    serializers.ModelSerializer
+):
+    site = serializers.StringRelatedField()
+    current_preview_task_status = serializers.SerializerMethodField()
+    latest_recalculation_result = serializers.SerializerMethodField()
+
+    def get_current_preview_task_status(self, obj):
+        async_result = AsyncResult(obj.task_id)
+        return async_result.status
+
+    def get_latest_recalculation_result(self, obj):
+        ordered_result = OrderedDict(
+            {
+                "unknown_character_count": obj.latest_recalculation_result[
+                    "unknown_character_count"
+                ],
+                "updated_entries": [],
```

Comment:
Maybe I'm missing something. Can this just be `"updated_entries": obj.latest_recalculation_result["updated_entries"]` and skip the transcribing below?

Reply:
The reason I'm transcribing is just for the proper order. The second OrderedDict is there to ensure the order of the "updated entry" fields are consistent. They sometimes get jumbled around in the JSONfield so the two ordered dicts ensure consistency with {"unknown_character_count", "updated_entries"} and {"title", "cleaned_title", is_title_updated", "previous_custom_order", "new_custom_order"} respectively